crates/proof-of-sql/src/base/database/column_operation.rs

```diff
@@ -256,6 +322,47 @@ where
         .collect()
 }
 
+/// Divide one slice by another of the same length, casting the left slice to the type of the right slice.
+///
+/// We do not check for length equality here
+pub(super) fn try_divide_slices_left_upcast<SmallerType, LargerType>(
+    lhs: &[SmallerType],
+    rhs: &[LargerType],
+) -> ColumnOperationResult<Vec<LargerType>>
+where
+    SmallerType: Copy + Debug,
+    LargerType: From<SmallerType> + CheckedDiv<Output = LargerType> + Copy + Debug,
+{
+    lhs.iter()
+        .zip(rhs.iter())
+        .map(|(l, r)| -> ColumnOperationResult<LargerType> {
+            LargerType::from(*l)
+                .checked_div(r)
+                .ok_or(ColumnOperationError::DivisionByZero)
+        })
+        .collect()
+}
+
+/// Divide one slice by another of the same length, casting the right slice to the type of the left slice.
+///
+/// We do not check for length equality here
+pub(super) fn try_divide_slices_right_upcast<SmallerType, LargerType>(
+    lhs: &[LargerType],
+    rhs: &[SmallerType],
+) -> ColumnOperationResult<Vec<LargerType>>
+where
+    SmallerType: Copy + Debug,
+    LargerType: From<SmallerType> + CheckedDiv<Output = LargerType> + Copy + Debug,
```

Comment:
NIT:
```suggestion
    SmallerType: Copy + Debug + Into<LargerType>,
    LargerType: CheckedDiv<Output = LargerType> + Copy + Debug,
```

Reply:
Right. We should implement `From` as opposed to `Into` but when it comes to type bounds we want to use `Into` instead since `From` implies `Into`.